10.1/root/usr/share/container-scripts/mysql/common.sh

```diff
@@ -162,3 +169,36 @@ function wait_for_mysql_master() {
     sleep 1
   done
 }
+
+# Converts string version to the integer format (5.5.33 is converted to 505,
+# 10.1.23-MariaDB is converted into 1001, etc.
+function version2number() {
+  local version_major=${1%.*}
+  printf %d%02d ${version_major%%.*} ${version_major##*.}
+}
```

Comment:
Agreed. Solution `version_major=$(echo "$1" | grep -o -e '^[0-9]*\.[0-9]*')` looks simplest from what I was thinking about, but if anybody has an idea how to make it correct enough without a fork, let me know..